sourcecode/apis/common/app/ApiModels/Resource.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\ApiModels;
+
+use App;
+use App\Apis\ResourceApiService;
+
+class Resource
+{
+    public string $id;
+    public string $resourceGroupId;
+    public ?string $deletedReason;
+    public ?string $deletedAt;
+    public string $updatedAt;
+    public string $createdAt;
+
+    public function __construct($data)
```

Comment:
I suggest an alternative approach here and in ResourceVersion: use promoted properties and call the constructor with the splat operator in front of the data array.

For instance:

~~~php
public function __construct(
    public string $id,
    public ?string $anOptionalParam = null,
    // etc.
) {
}

// In the service class:
new Resource(...$data);
~~~

This will map the array elements to each property by name, avoiding the need to manually assign them, and letting the IDE provide better help when an instance is constructed without the use of an array (e.g. in unit tests).